main.py

```diff
@@ -1,52 +1,70 @@
-    default_log='None'
-    def __init__(self,user_name,log=default_log):
-        self.user_name=user_name
-        self.log=str(log)
+class User():
+    def __init__(self,name):
+        self.name=name
         self.card=Card()
+
     def info(self):
-        print(f'User name:{self.user_name}')
-        return (self.card.info_card())
+        print(f'User name:{self.name}')
+        return (self.card.info())
+
+    def logg(self):
+        return(self.card.loggering())
 
 class Card:
     default_ballance=0
-    default_uuid=uuid.uuid4()
-    def __init__(self,uuid=default_uuid,ballance=default_ballance):
-        self.uuid=uuid
+    default_log=[]
+
+    def __init__(self,ballance=default_ballance,log=default_log):
         self.ballance=float(ballance)
+        self.log=log
+
+
+    def loggering(self):
+        # log function, main problem: logging at one log for two users at same time
+        if self.log==[]:
+            print("You haven't any activity")
+        else:
+            print(f'Your log of operations:')
+            for i in self.log:
+                print(i)
 
-    def info_card(self):
-        #Метод виводу інформації по карті
+    def info(self):
+        # Print user's card general information
         print(f'You hawe: {self.ballance} usd money')
-        print(f'Your card number:{self.uuid}')
+        print(f'Your card number:{uuid.uuid4()}')
+        print(f'You have:{len(self.log)} transaction')
+
 
-    def earn_money(self,value):
-        #метод ініціалізаціі певної стартової сумі коштив на карту
+    def earn_money(self,value,trans):
+        #проблема з записом в лог - записи дублює в лог обох коростувачів, або ж один лог вийшов на двох
         if value>0:
             self.ballance+=value
+            self.log.append(
+                f'You resive {value} usd in your acount, from {trans.name} acoun total ballance is: {self.ballance}')
         else:
             print('Error: you cant earn this shit!')
 
 
-    def transaction(self,value,recipient):
-        #метод проведення транзакції з карти на отримувача
+    def transaction(self,value,recipient,trans):
+        # Transaction function with loggering in both logs (self and recipient)
         if value<=self.ballance:
             self.ballance-=value
-            recipient.card.earn_money(value)
-
-            print(f'You hawe transacted {value} usd,from your card, to the {recipient} card, total ballance is: {self.ballance} ')
+            recipient.card.earn_money(value,trans)
+            self.log.append(f'You have transacted {value} usd,from your acount, to the {recipient.name} acount, total ballance'
+                            f' is: {self.ballance} usd. ')
         else:
-            print(f'You hawe not inought money')
+            print(f"You dont have enought money")
             
 
-fedor=Bank('Fedor_Ovchinkin')
-nikon=Bank('Nikodim_Rozetkin')
-fedor.info()
+fedor=User('Fedor_Ovchinkin')
+nikon=User('Nikodim_Rozetkin')
+bank=User('Bank')
```

Comment:
is bank a user? (seems semantically wrong for me)
why do you need it here at all?